File: Tugas Besar/Gudang.php
<?php
//Simpanlah dengan nama file : Gudang.php
require_once 'database.php';
class Gudang 
{
    private $db;
    private $table = 'gudang';
    public $nomergudang = "";
    public $kodebarang = "";
    public $nama_barang = "";
    public $stok_barang = "";
    public function __construct(MySQLDatabase $db)
    {
        $this->db = $db;
    }
    public function get_all() 
    {
        $query = "SELECT * FROM $this->table";
        $result_set = $this->db->query($query);
        return $result_set;
    }
    public function get_by_id(int $id)
    {
        $query = "SELECT * FROM $this->table WHERE id = $id";
        $result_set = $this->db->query($query);   
        return $result_set;
    }
    public function get_by_nomergudang(int $nomergudang)
    {
        $query = "SELECT * FROM $this->table WHERE nomergudang = $nomergudang";
        $result_set = $this->db->query($query);   
        return $result_set;
    }
    public function insert(): int
    {
        $query = "INSERT INTO $this->table (`nomergudang`,`kodebarang`,`nama_barang`,`stok_barang`) VALUES ('$this->nomergudang','$this->kodebarang','$this->nama_barang','$this->stok_barang')";
        $this->db->query($query);
        return $this->db->insert_id();
    }
    public function update(int $id): int
    {
        $query = "UPDATE $this->table SET nomergudang = '$this->nomergudang', kodebarang = '$this->kodebarang', nama_barang = '$this->nama_barang', stok_barang = '$this->stok_barang' 
        WHERE id = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function update_by_nomergudang($nomergudang): int
    {
        $query = "UPDATE $this->table SET nomergudang = '$this->nomergudang', kodebarang = '$this->kodebarang', nama_barang = '$this->nama_barang', stok_barang = '$this->stok_barang' 
        WHERE nomergudang = $nomergudang";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function delete(int $id): int
    {
        $query = "DELETE FROM $this->table WHERE id = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function delete_by_nomergudang($nomergudang): int
    {
        $query = "DELETE FROM $this->table WHERE nomergudang = $nomergudang";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
}
?>

File: Tugas Besar/Gudang.py
import requests
import json
class Gudang:
    def __init__(self):
        self.__id=None
        self.__nomergudang = None
        self.__kodebarang = None
        self.__nama_barang = None
        self.__stok_barang = None
        self.__url = "http://localhost/Penjualan_api/gudang_api.php"
                    
    @property
    def id(self):
        return self.__id
    @property
    def nomergudang(self):
        return self.__nomergudang
        
    @nomergudang.setter
    def nomergudang(self, value):
        self.__nomergudang = value
    @property
    def kodebarang(self):
        return self.__kodebarang
        
    @kodebarang.setter
    def kodebarang(self, value):
        self.__kodebarang = value
    @property
    def nama_barang(self):
        return self.__nama_barang
        
    @nama_barang.setter
    def nama_barang(self, value):
        self.__nama_barang = value
    @property
    def stok_barang(self):
        return self.__stok_barang
        
    @stok_barang.setter
    def stok_barang(self, value):
        self.__stok_barang = value
    def get_all(self):
        payload ={}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(self.__url, json=payload, headers=headers)
        return response.text
    def get_by_nomergudang(self, nomergudang):
        url = self.__url+"?nomergudang="+nomergudang
        payload = {}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(url, json=payload, headers=headers)
        data = json.loads(response.text)
        for item in data:
            self.__id = item['id']
            self.__nomergudang = item['nomergudang']
            self.__kodebarang = item['kodebarang']
            self.__nama_barang = item['nama_barang']
            self.__stok_barang = item['stok_barang']
        return data
    def simpan(self):
        payload = {
            "nomergudang":self.__nomergudang,
            "kodebarang":self.__kodebarang,
            "nama_barang":self.__nama_barang,
            "stok_barang":self.__stok_barang
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.post(self.__url, data=payload, headers=headers)
        return response.text
    def update_by_nomergudang(self, nomergudang):
        url = self.__url+"?nomergudang="+nomergudang
        payload = {
            "nomergudang":self.__nomergudang,
            "kodebarang":self.__kodebarang,
            "nama_barang":self.__nama_barang,
            "stok_barang":self.__stok_barang
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.put(url, data=payload, headers=headers)
        return response.text
    def delete_by_nomergudang(self,nomergudang):
        url = self.__url+"?nomergudang="+nomergudang
        headers = {'Content-Type': 'application/json'}
        payload={}
        response = requests.delete(url, json=payload, headers=headers)
        return response.text
